isminet/models/network.py

```
@@ -0,0 +1,121 @@
+        Validates the DHCP range configuration when server mode is enabled.
+
+        Ensures that both start and end IP addresses are specified when DHCP server mode is active.
+
+        Raises:
+            ValueError: If DHCP server mode is enabled but start or end IP addresses are not set.
+
+        Returns:
+            DHCPConfiguration: The validated configuration instance.
+        """
+        if self.mode == DHCPMode.SERVER and self.enabled:
+            if not all([self.start, self.end]):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)